source/base/parameter_handler.cc

```diff
@@ -2969,83 +2890,59 @@ ParameterHandler::scan_line (std::string         line,
-                    << "    " << entry_name << std::endl
-                    << "    <Present subsection:" << std::endl;
-          for (unsigned int i=0; i<subsection_path.size(); ++i)
-            std::cerr << std::setw(i*2+8) << " "
-                      << "subsection " << subsection_path[i] << std::endl;
-          std::cerr << "    >" << std::endl;
-
-          return false;
+          Assert(false,
+                 ExcCannotParseLine(current_line_n,
+                                    input_filename,
+                                    ("No entry with name <" + entry_name +
+                                     "> was declared in the current subsection.")));
         }
     }
-
   // an include statement?
-  if ((line.find ("INCLUDE ") == 0) ||
-      (line.find ("include ") == 0))
+  else if ((line.find ("INCLUDE ") == 0) ||
+           (line.find ("include ") == 0))
     {
       // erase "include " statement and eliminate spaces
       line.erase (0, 7);
       while ((line.size() > 0) && (line[0] == ' '))
         line.erase (0, 1);
 
       // the remainder must then be a filename
-      if (line.size() == 0)
-        {
-          std::cerr << "Line <" << current_line_n
-                    << "> of file <" << input_filename
-                    << "> is an include statement but does not name a file!"
-                    << std::endl;
-
-          return false;
-        }
+      AssertThrow (line.size() !=0,
+                   ExcCannotParseLine (current_line_n,
+                                       input_filename,
+                                       "The current line is an 'include' or "
+                                       "'INCLUDE' statement, but it does not "
+                                       "name a file for inclusion."));
 
       std::ifstream input (line.c_str());
-      if (!input)
-        {
-          std::cerr << "Line <" << current_line_n
-                    << "> of file <" << input_filename
-                    << "> is an include statement but the file <"
-                    << line << "> could not be opened!"
-                    << std::endl;
-
-          return false;
-        }
-      else
-        return read_input (input);
+      AssertThrow (input, ExcCannotOpenIncludeStatementFile (current_line_n,
+                                                             input_filename,
+                                                             line));
+      read_input (input);
+    }
+  else
+    {
+      AssertThrow (false,
+                   ExcCannotParseLine (current_line_n,
+                                       input_filename,
+                                       "This line could not be parsed: please check "
+                                       "to make sure that you did not miss a 'set', "
+                                       "'include', 'subsection', or 'end' statement "
+                                       "here."));
```

Comment:
I think it's not quite clear to me what you mean with "miss" here. Maybe say that the only valid statements are 'set', 'include', 'subsection' and 'end'?